common/src/main/java/com/thoughtworks/go/server/domain/AgentInstances.java

```diff
@@ -40,15 +40,15 @@
 
     private SystemEnvironment systemEnvironment;
     private Map<String, AgentInstance> agentInstances = new ConcurrentHashMap<>();
-    private AgentRuntimeStatus.ChangeListener changeListener;
+    private AgentStatusChangeListener agentStatusChangeListener;
 
-    public AgentInstances(AgentRuntimeStatus.ChangeListener changeListener) {
-        this.changeListener = changeListener;
+    public AgentInstances(AgentStatusChangeListener agentStatusChangeListener) {
+        this.agentStatusChangeListener = agentStatusChangeListener;
         this.systemEnvironment = new SystemEnvironment();
     }
 
-    public AgentInstances(AgentRuntimeStatus.ChangeListener changeListener, SystemEnvironment systemEnvironment, AgentInstance... agentInstances) {
-        this(changeListener);
+    public AgentInstances(SystemEnvironment systemEnvironment, AgentInstance... agentInstances) {
+        this(null);
```

Comment:
Since this constructor is being used only in tests so far, its fine to pass along `null` for `AgentStatusChangeListener`, but could you instead pass along that `null` value from tests instead of hardcoding that in the constructor here.

Reply:
Done